pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -167,19 +171,25 @@ private SegmentNameGenerator getSegmentNameGenerator(SegmentGeneratorConfig segm
         return new InputFileSegmentNameGenerator(segmentNameGeneratorConfigs.get(FILE_PATH_PATTERN),
             segmentNameGeneratorConfigs.get(SEGMENT_NAME_TEMPLATE), inputFileUri, appendUUIDToSegmentName);
       case BatchConfigProperties.SegmentNameGeneratorType.UPLOADED_REALTIME:
-        Preconditions.checkState(segmentGeneratorConfig.getCreationTime() != null,
+        Preconditions.checkState(segmentNameGeneratorConfigs.get(SEGMENT_UPLOAD_TIME_MS) != null,
             "Creation time must be set for uploaded realtime segment name generator");
-        Preconditions.checkState(segmentGeneratorConfig.getUploadedSegmentPartitionId() != -1,
+        Preconditions.checkState(segmentNameGeneratorConfigs.get(SEGMENT_PARTITION_ID) != null,
             "Valid partition id must be set for uploaded realtime segment name generator");
         long creationTime;
```

Comment:
Suggest renaming this as it is upload time instead of creation time.

Reply:
Sure, but I'm curious if this even matters? I think only partition id and table name actually matters for upsert table

Reply:
It matters to break tie when 2 segments contain the records with same timestamp (comparison value)

Reply:
but for upsert table same comparison value means same row, query should return same result